pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/PublishDraftVersion.java

```diff
@@ -114,46 +129,145 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
+                .toChild(sourceContent -> sourceContent.released())
+                .getAsOptional();
+        if (draftSource != null) {
+            if (releasedSource.isPresent()) {
+                try {
+                    releasedSource.get().delete();
+                } catch (PersistenceException e) {
+                    throw new RuntimeException("Failed to remove source/released: " + releasedSource.get().getPath());
+                }
+            }
+            try {
+                rename(draftSource, "released");
+            } catch (RepositoryException e) {
+                throw new RuntimeException("Cannot find source/draft: " + draftSource.getPath());
+            }
+
+    }
+    }
+
+    /**
+     *  Method to check if request contains an assembly and it needs to get published
+     *
+     * @param request
+     * @param response
+     * @param changes
+     * @param serviceResourceResolver
+     * @return
+     * @throws PersistenceException
+     */
+    private void publishAssembly(SlingHttpServletRequest request, PostResponse response, List<Modification> changes, ResourceResolver serviceResourceResolver) throws PersistenceException {
+            Assembly assembly = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Assembly.class);
             Locale locale = getLocale(request);
             String variant = getVariant(request);
             // Get the draft version, there should be one
-            Optional<ModuleVersion> versionToRelease = module.getDraftVersion(locale, variant);
+            Optional<AssemblyVersion> versionToRelease = assembly.getDraftVersion(locale, variant);
             if (!versionToRelease.isPresent()) {
                 response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
                         "The module doesn't have a draft version to be released");
-                return;
-            } else if (versionToRelease.get().metadata().getOrCreate().productVersion().get() == null
+                return ;
+            }
+            else if (versionToRelease.get().metadata().getOrCreate().productVersion().get() == null
                     || versionToRelease.get().metadata().getOrCreate().productVersion().get().isEmpty()) {
                 // Check if productVersion is set
                 response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
                         "The version to be released doesn't have productVersion metadata");
-                return;
+                return ;
             } else if (versionToRelease.get().metadata().getOrCreate().urlFragment().get() == null
                     || versionToRelease.get().metadata().getOrCreate().urlFragment().get().isEmpty()) {
                 // Check if urlFragment is set
                 response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
                         "The version to be released doesn't have urlFragment metadata");
```

Comment:
urlFragment is an optional field

Reply:
@xdavidson, urlFragment is not a part of this ticket.

Reply:
Agreed. But it would be super awesome if you can remove the validation on requiring urlFragment.